freqtrade/rpc/rpc.py

```diff
@@ -1035,16 +1035,49 @@ def _convert_dataframe_to_dict(strategy: str, pair: str, timeframe: str, datafra
 
     def _rpc_analysed_dataframe(self, pair: str, timeframe: str,
                                 limit: Optional[int]) -> Dict[str, Any]:
+        """ Analyzed dataframe in Dict form """
 
+        _data, last_analyzed = self.__rpc_analysed_dataframe_raw(pair, timeframe, limit)
+        return self._convert_dataframe_to_dict(self._freqtrade.config['strategy'],
+                                               pair, timeframe, _data, last_analyzed)
+
+    def __rpc_analysed_dataframe_raw(self, pair: str, timeframe: str,
+                                     limit: Optional[int]) -> Tuple[DataFrame, datetime]:
+        """ Get the dataframe and last analyze from the dataprovider """
         _data, last_analyzed = self._freqtrade.dataprovider.get_analyzed_dataframe(
             pair, timeframe)
         _data = _data.copy()
+
         if limit:
             _data = _data.iloc[-limit:]
-        return self._convert_dataframe_to_dict(self._freqtrade.config['strategy'],
-                                               pair, timeframe, _data, last_analyzed)
-
-    @staticmethod
+        return _data, last_analyzed
+
+    def _ws_all_analysed_dataframes(
+        self,
+        pairlist: List[str],
+        limit: Optional[int]
+    ) -> Dict[str, Any]:
+        """ Get the analysed dataframes of each pair in the pairlist """
+        timeframe = self._freqtrade.config['timeframe']
+        candle_type = self._freqtrade.config.get('candle_type_def', CandleType.SPOT)
+        _data = {}
+
+        for pair in pairlist:
+            dataframe, last_analyzed = self.__rpc_analysed_dataframe_raw(pair, timeframe, limit)
+            _data[pair] = {"key": (pair, timeframe, candle_type), "value": dataframe}
```

Comment:
I think the format for this (basically what we're sending) should change (obviously not just here - but also for realtime messages).
We're currently ignoring `last_analyzed` - but that's a quite important point for the receiver (it'll know how old the analyzed dataframe is).

we can't assume that the last candle date is the analyzed data - there may be delays (maybe it's a slow strategy that takes 5 minutes to go through the full pairlist).
While it's not a problem - it will be a useful information on the receiver. We're currently use `datetime.now()` in dataprovider when adding the dataframe - but when we received the pair is irrelevant - what counts is when it was analyzed.

Reply:
This has been added :+1: